src/anemoi/models/models/encoder_processor_decoder.py

```diff
@@ -58,48 +63,50 @@ def __init__(
 
         self._define_tensor_sizes(config)
 
-        # Create trainable tensors
-        self._create_trainable_attributes()
-
         # Register lat/lon
-        self._register_latlon("data", self._graph_name_data)
-        self._register_latlon("hidden", self._graph_name_hidden)
+        for name in self._graph_mesh_names:
+            self._register_latlon(name)
 
         self.num_channels = config.model.num_channels
 
-        input_dim = self.multi_step * self.num_input_channels + self.latlons_data.shape[1] + self.trainable_data_size
+        input_dim = self.multi_step * self.num_input_channels
 
         # Encoder data -> hidden
-        self.encoder = instantiate(
-            config.model.encoder,
-            in_channels_src=input_dim,
-            in_channels_dst=self.latlons_hidden.shape[1] + self.trainable_hidden_size,
-            hidden_dim=self.num_channels,
-            sub_graph=self._graph_data[(self._graph_name_data, "to", self._graph_name_hidden)],
-            src_grid_size=self._data_grid_size,
-            dst_grid_size=self._hidden_grid_size,
-        )
+        self.encoders = nn.ModuleDict()
+        for data in self._graph_input_meshes:
+            self.encoders[data] = instantiate(
+                config.model.encoder,
+                in_channels_src=input_dim + self.num_node_features[data] + self.num_trainable_params[data],
+                in_channels_dst=self.num_node_features[self._graph_name_hidden]
+                + self.num_trainable_params[self._graph_name_hidden],
+                hidden_dim=self.num_channels,
+                sub_graph=self._graph_data[(data, "to", self._graph_name_hidden)],
+                src_grid_size=self.num_nodes[data],
+                dst_grid_size=self.num_nodes[self._graph_name_hidden],
+            )
 
         # Processor hidden -> hidden
         self.processor = instantiate(
             config.model.processor,
             num_channels=self.num_channels,
-            sub_graph=self._graph_data[(self._graph_name_hidden, "to", self._graph_name_hidden)],
-            src_grid_size=self._hidden_grid_size,
-            dst_grid_size=self._hidden_grid_size,
+            sub_graph=self._graph_data.get((self._graph_name_hidden, "to", self._graph_name_hidden), None),
```

Comment:
What happens if `None` is passed to the `sub_graph`?

Reply:
It will fail unless the TransformerProcessor is specified.